src/main/capi/pending-c.cpp

```diff
@@ -82,12 +83,12 @@ duckdb_pending_state duckdb_pending_execute_task(duckdb_pending_result pending_r
 	PendingExecutionResult return_value;
 	try {
 		return_value = wrapper->statement->ExecuteTask();
-	} catch (const duckdb::Exception &ex) {
```

Comment:
There's a reason we catch both `duckdb::Exception` and `std::exception` here - `PreservedError` looks at the exception type from `duckdb::Exception` and uses that to print the error type prefix (e.g. `Catalog Error: ...`).

I'm actually removing all of this in my exception refactor - but currently removing this branch changes behavior